docs/e2e-test.md

```diff
@@ -0,0 +1,44 @@
+# Ramen End-to-End (E2E) Test Framework
+
+This document will provide an introduction to the framework, describe its components and structure, and explain how to run the tests.
+
+## Overview
+
+The E2E test framework is designed to validate the functionality of the Ramen project from a user's perspective. This means that it will test the entire system. The framework is built using Go and follows the native Go test format. Setting up the test environment is outside the scope of the framework, and is left to the user. The framework will however, detect the environment and run the tests accordingly.
+
+## Structure
+
+### Go workspaces
+
+The E2E test framework uses Go workspaces to separate the test dependencies from the Ramen dependencies. This is done to avoid any conflicts between the two. The Go workspaces are defined in the go.work file, and the checksums for the dependencies are defined in the go.work.sum file. Go workspaces provides us the following benefits:
+
+* Isolation of Test Dependencies: By using a Go.work file, we can maintain a separate list of dependencies for our tests. This approach keeps our main project's dependencies uncluttered, reducing the risk of conflicts and making it easier to manage dependencies.
+* Improved Reproducibility: A Go.work file allows us to explicitly define the versions of dependencies required for our tests. This ensures that all team members work with the same dependency versions, leading to consistent test results and reducing the likelihood of hard-to-debug issues stemming from dependency mismatches.
+* Simplified Dependency Updates: When updating dependencies, a Go.work file enables us to quickly identify and update test-specific dependencies without affecting the main project. This reduces the chances of inadvertently introducing breaking changes or compatibility issues.
+* Easier Onboarding: For new team members or external contributors, a Go.work file provides a clear separation of test dependencies from the main project. This simplifies the onboarding process and makes it easier for newcomers to understand the project structure and contribute effectively.
+* Enhanced Security: By isolating test dependencies, we minimize the potential attack surface in our main project. This approach helps ensure that vulnerabilities in test dependencies do not compromise the security of our production code.
```

Comment:
Isolating the e2e framework from ramen is nice - but we do really need go workspace? Can just have a separate go project in the e2e directory? Go will treat such project as part of ramen project?

Regardless, this text is really not useful for developers, so we can move it to the end of the doc.

The most important thing to be at the top is how to install this (hopefully nothing needed) and how to run the tests. Everything else can be at the end of the doc or in another doc.